README.md

```diff
@@ -0,0 +1,222 @@
+* Book Management (CRUD operations, searching)
+* Category Management (CRUD operations)
+* Shopping Cart Management (adding, updating, removing items)
+* Order Management (placing orders, viewing order history, order details)
+* User Role Management (Admin vs. User functionalities)
+* Unit & Integration API Testing (ensuring code reliability)
+
+## 🔨 Technologies
+
+* **Spring Boot**: framework for building Java-based web applications
+* **Spring Security**: security and user management
+* **Spring Data JPA**: data persistence
+* **Swagger**: API documentation
+* **Docker**: deployment and scalability
+* **Liquibase**: database management
+* **MySQL**: database management system
+* **Mockito**: mocking frameworks for tests
+* **JUnit**: testing and reliability
+* **Testcontainers**: in-memory database for integration testing
+
+## 🤸 Installation without Docker
+<details>
+<summary>Click for detailed instructions</summary>
+
+### 1. Prerequisites
+
+* Java 21+
+* Maven 4+
+* MySQL 8+
+
+### 2. Clone the repository
+
+```bash
+git clone https://github.com/makaroshyna/online-book-store.git
+cd online-book-store
+```
+
+### 3. Set up MySQL
+
+Establish a new MySQL database, recording its URL, username, and password for future reference.
+
+### 4. Configure environment variables
+
+Create a file names `application.properties` in `src/main/resources` with the following context
+(replace placeholders with your details):
+```properties
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+spring.datasource.url=jdbc:mysql://localhost:3306/your_db_name
+spring.datasource.username=your_db_username
+spring.datasource.password=your_db_password
+spring.jpa.hibernate.ddl-auto=update
+server.servlet.context-path=/api
+
+jwt.expiration=token_expiration_time
+jwt.secret=your_secret_key
+```
+
+### 5. Get set up
+
+Run the following command to install any required dependencies and build your project:
+```bash
+mvn clean install
+```
+
+### 6. Start the server
+
+Once the build is complete, use this command to run your application:
+```bash
+mvn spring-boot:run
+```
+Your server will be accessible at `http://localhost:8080`.
+
+</details>
+
+## 🤖 Installation with Docker
+
+<details>
+<summary>Click for detailed instructions</summary>
+  
+### 1. Prerequisites
+
+* Docker
+
+### 2. Clone the repository
+
+```bash
+git clone https://github.com/makaroshyna/online-book-store.git
+cd online-book-store
+```
+
+### 3. Configure environment variables
+
+Create an .env file in the project root directory and add the following:
+```env
+MYSQLDB_DATABASE=your_db_name
+MYSQLDB_USER=your_db_user
+MYSQLDB_PASSWORD=your_dbpassword
+MYSQLDB_ROOT_PASSWORD=your_db_root_password
+
+MYSQLDB_LOCAL_PORT=3306
+MYSQLDB_DOCKER_PORT=3306
+
+SPRING_LOCAL_PORT=8080
+SPRING_DOCKER_PORT=8080
+DEBUG_PORT=5005
+```
+
+### 4. Install dependencies and build the project
+
+```bash
+mvn clean install
+```
+
+### 5. Build and run the Docker containers
+
+```bash
+docker compose up
```

Comment:
I think we can remove the point above by adding this
```suggestion
docker compose up --build
```
details: https://docs.docker.com/reference/cli/docker/compose/up/
BTW did you manage to resolve the docker problem?

Reply:
Nope, still have it :(